waku/v2/node/waku_node.nim

```diff
@@ -113,16 +113,22 @@ proc getAutonatService*(rng: ref HmacDrbgContext): AutonatService =
   let autonatService = AutonatService.new(
     autonatClient = AutonatClient.new(),
     rng = rng,
-    scheduleInterval = some(chronos.seconds(120)),
+    scheduleInterval = Opt.some(chronos.seconds(120)),
     askNewConnectedPeers = false,
     numPeersToAsk = 3,
     maxQueueSize = 3,
     minConfidence = 0.7)
 
-  proc statusAndConfidenceHandler(networkReachability: NetworkReachability, confidence: Option[float]) {.gcsafe, async.} =
+  proc statusAndConfidenceHandler(networkReachability: NetworkReachability,
+                                  confidence: Opt[float]):
+                                  Future[void]  {.gcsafe, raises: [].} =
     if confidence.isSome():
       info "Peer reachability status", networkReachability=networkReachability, confidence=confidence.get()
 
+    let retFut = newFuture[void]()
+    retFut.complete()
+    return retFut
+
```

Comment:
```suggestion
  proc statusAndConfidenceHandler(networkReachability: NetworkReachability,
                                  confidence: Opt[float]):
                                  Future[void]  {.gcsafe, async.} =
    if confidence.isSome():
      info "Peer reachability status", networkReachability=networkReachability, confidence=confidence.get()
```

Sorry if I wasn't clear on discord, that is what I meant when I said "Some of the failures are because you removed the async from statusAndConfidenceHandler"

Reply:
Thanks for the comment @Menduist ! It is much better that way!